include/spm.h

```diff
@@ -41,6 +48,25 @@ void spm_jump(void);
  */
 void spm_config(void);
 
+/** @brief Set handler which is called on the fault handler on the secure parition.
+ *
+ * Handler is intended to be used to print out any pending log data before reset.
+ *
+ * @note It is only for debugging purposes!
+ *
+ * @param handler Handler.
+ *
+ * @retval -ENOTSUP if feature is disabled.
+ * @retval 0 on success.
+ */
+int spm_ns_set_fatal_error_handler(spm_ns_on_fatal_error_t handler);
```

Comment:
Do you think that `spm_set_ns_fatal_error_handler` is a better name? Currently it sounds like you are setting the error handler of the non-secure spm.